### DISCUSSION TITLE
Playlist with Live AND VoD

### DISCUSSION THREAD

**Discussion by mistermarti** [ant-media/Ant-Media-Server#5599] [Q&A]
Hi all, 

I know there's a way to create linear live streaming:
https://antmedia.io/docs/guides/publish-live-stream/Playlist/

But this function apply only to VOD files.

In my case, I'd need to get the same linear live, but as we can do it in wowza with .smil files, with a list of LIVE and VODs url and files, each with specific duration (especially for live).

Does anybody knows how to workaround it, or if there's any function in development allowing this?

Thanks and have a nice day,

Nicolas

**Comment by Mohit-3196** [comment]
Hi @mistermarti ,
Hope you are doing well.

Yes, the playlist feature is for VOD files, and with that you can also append new VOD files into the playlist on the FLY
https://github.com/orgs/ant-media/discussions/4903#discussioncomment-5338039

In your case, you would like to switch between Playlist and Live for the same stream with the same streamId without having to stop the stream. Is that right?

Thank you

**Reply by mistermarti** [reply]
Hi @Mohit-3196,

Thanks for your reply!

yes, exactly, switch between different vod files, and live feed. I’d need also to be able to give a duration for the live, for example:

1. vod01 
2. Live - 5 minutes
3. Vod02
4. Live - 5 minutes

all that in loop.

there maybe some scripting to do for that ?

thanks, have a good day!


Nicolas

**Reply by Mohit-3196** [reply]
Hi @mistermarti,
These things should be doable via some plugin by utilizing REST API calls and webrtc_adaptors

Right now on the server side, there is not a direct implementation to use it as it is for this use case. Still, we had a similar request sometime back and we delegated it to the community/developers to make such an implementation for us. Me and my colleague @NurC123 were following that. 

So would you be willing to such a solution and contribute towards the development? If yes, we can just revisit that thread and see how we can move ahead.

Thank you

**Reply by mistermarti** [reply]
Hi @Mohit-3196, thanks for your feedback!
This would be really interesting, yes, if the developper/community would be able to work on it?
I'm new in this environment and not sure about what this means - in terms of contributions. Does this means implications in terms of project management, tests, ressources sharing? Or payment of ressources?
Thanks a lot for telling me what is possible, and how.
All the best,

Nicolas

**Reply by Mohit-3196** [reply]
Hi @mistermarti,
It's great to hear back from you.

> I'm new in this environment and not sure about what this means - in terms of contributions. Does this means implications in terms of project management, tests, ressources sharing? Or payment of resources?

There are many ways in which you can contribute towards this. It can be via making some development if you can or by sharing the resources and of course, we'll be together in this to guide/help you whenever needed.
We can share the possible ways to do this and you can integrate those together into some sort of plugin moving forward the same can be also added to the [AMS Marketplace](https://antmedia.io/marketplace/) so that you can monetize the same. I'm just thinking out loud and sharing with you :)

Alternatively, if you feel that there are some limitations to devoting such sort of time, we can delegate it to the community/developers and you can contribute towards the development fee.
I mean there are some options and we can discuss and proceed in a manner that suits you the most.

Thank you

**Reply by mekya** [reply]
Hi @Mohit-3196 and @mistermarti ,

There is an issue here, and I've recommended a solution. Could you please comment there?

https://github.com/ant-media/Ant-Media-Server/issues/5732

Cheers

**Reply by Mohit-3196** [reply]
Thank you @mekya abi for the suggestions/feedforward.
I think they are good. @mistermarti can comment further.

Thank you.

**Reply by mistermarti** [reply]
Hi @Mohit-3196 and @mekya sorry for my silence :-( thanks a lot for your propositions and answers.

What I can tell here are the followings:

- I don't have developper level in order to integrate what you would explain me and test
- Of course, I would take time to work with the community and developers and test with them, participate at the project management level
- I don't imagine and don't know how that is working when we ask the community/developers to do some work

This would depend on the amount, and how it is working, for me and the community.

If you can give me some ideas about this, it would clarify my final choice.

Again, thanks for your help and kind regards,

Nicolas

**Reply by Mohit-3196** [reply]
Hi @mistermarti,
How are you doing?

No worries :) It happens at times.

One of the senior developers reviewed this request and suggested a couple of solutions.

As per the first suggestions, as a workaround solution the below steps can be utilized.

> Vod01
> Live01 (Let the Ant Media Server pull Live HLS internally as stream source: http://ant-media-server/WebRTCAppEE/streams/live01.m3u8)
> Vod02
> Live02 (Let the Ant Media Server pull Live HLS internally as stream source: http://ant-media-server/WebRTCAppEE/streams/live02.m3u8)

In case the above don't work for you, we can make some developments and create a solution as below.
 
> We provide a REST method to skip the items, such as play next item (i.e., switch from Live01 to VoD02)
> We provide a duration field for the live streams. I mean, it may play Live01 for 5 minutes(configurable) and proceeds to next item after 5 minutes, no matter if live streams continue.
> Please let us know how you would like to proceed.

So, I request you to review these once and let us know your observations/feedforwards here https://github.com/ant-media/Ant-Media-Server/issues/5732

Let me answer your other questions as well.

> I don't have developper level in order to integrate what you would explain me and test.

I think it's not a big issue or a blocker. We can help you with your questions and make some sort of samples as well and in case you need further help, you can get a dedicated support package.

> Of course, I would take time to work with the community and developers and test with them, participate at the project management level

Perfect, that's good to know.

> I don't imagine and don't know how that is working when we ask the community/developers to do some work

Actually the process is very transparent about this.
Once someone requests a feature/developemt from the community then anyone from community developers pick it up and work on it. Ofcourse there can be some monetary quotations for that and the requster and the developer can decide it mutually.
Also, if it is something that can be made as a plugin or application, it can be listed on the AMS Marketplace as well and can be monetized further.

Thank you.

**Reply by mistermarti** [reply]
Dear @Mohit-3196,

Many thanks for your great explanations. All is good, thanks, hope the same for you!

Ok let me check in the upcoming days the workaround if this works. Not sure about the time allowed for each Live, i.e. 2 minutes Live from Live camera, and 30 seconds for Live from VOD file.

I'll then write my observations in #5732.

Really nice to ear about the features requests. This can be done after this first part.

Again, thanks for your help and I wish you a nice week-end!

Nicolas
